src/Utility/TypeExtensions.cs

```diff
@@ -142,7 +142,7 @@ public static object ConvertFromJson(string json)
         private static string ConvertToJson(object fromObj)
         {
             var context = new JsonObject.ConvertToJsonContext(
-                maxDepth: 4,
+                maxDepth: 50,
```

Comment:
We need this change because the "fast" Durable Functions OOProc model utilizes deeply nested JSON objects to represented WhenAll and WhenAny actions. I'm happy to discuss just how deep we really need this to be, but 4 is certainly too low for even the simplest WhenAny/WhenAll JSON.

Reply:
Could you please add a comment on what 50 means in this context?

Reply:
Copy-pasting from a related comment: 

I added a comment. The gist of it is that Durable Functions, irrespective of the new external package we're developing, may produce deeply nested JSON objects. Therefore, a depth of 4 was fairly limited and error prone for durable users.

Reply:
I chose 50 arbitrarily. I just wanted to make it high enough to cover most durable cases. In theory, I'd be ok with setting it even higher.

Reply:
Make sense, thank you for the detailed explanation.